app/models/comment.rb

```diff
@@ -55,6 +56,232 @@ def akismet_attributes
+        users << self.comment_owner
+      end
+      if notify_user_by_email?(self.comment_owner) && notify_user_of_own_comments?(self.comment_owner)
+        CommentMailer.comment_sent_notification(self).deliver
+      end
+
+      # send notification to the owner(s) of the ultimate parent, who can be users or admins
+      if self.ultimate_parent.is_a?(AdminPost)
+        AdminMailer.edited_comment_notification(self.id).deliver
+      else
+        # at this point, users contains those who've already been notified
+        if users.empty?
+          users = self.ultimate_parent.commentable_owners
+        else
+          # replace with the owners of the commentable who haven't already been notified
+          users = self.ultimate_parent.commentable_owners - users
+        end
+        users.each do |user|
+          unless user == self.comment_owner && !notify_user_of_own_comments?(user)
+            if notify_user_by_email?(user) || self.ultimate_parent.is_a?(Tag)
+              CommentMailer.edited_comment_notification(user, self).deliver
+            end
+            if notify_user_by_inbox?(user)
+              update_feedback_in_inbox(user)
+            end
+          end
+        end
+      end
+
+    end
+  end
+
+  after_create :after_create
+  def after_create
+    self.reload
+    # eventually we will set the locale to the user's stored language of choice
+    #Locale.set ArchiveConfig.SUPPORTED_LOCALES[ArchiveConfig.DEFAULT_LOCALE]
+    users = []
+    admins = []
+
+    # notify the commenter
+    if self.comment_owner && notify_user_of_own_comments?(self.comment_owner)
+      users << self.comment_owner
+    end
+    if notify_user_by_email?(self.comment_owner) && notify_user_of_own_comments?(self.comment_owner)
+      CommentMailer.comment_sent_notification(self).deliver
+    end
+
+    # Reply to owner of parent comment if this is a reply comment
+    if (parent_comment_owner = notify_parent_comment_owner)
+      users << parent_comment_owner
+    end
+
+    # send notification to the owner(s) of the ultimate parent, who can be users or admins
+    if self.ultimate_parent.is_a?(AdminPost)
+      AdminMailer.comment_notification(self.id).deliver
+    else
+      # at this point, users contains those who've already been notified
+      if users.empty?
+        users = self.ultimate_parent.commentable_owners
+      else
+        # replace with the owners of the commentable who haven't already been notified
+        users = self.ultimate_parent.commentable_owners - users
+      end
+      users.each do |user|
+        unless user == self.comment_owner && !notify_user_of_own_comments?(user)
+          if notify_user_by_email?(user) || self.ultimate_parent.is_a?(Tag)
+            CommentMailer.comment_notification(user, self).deliver
+          end
+          if notify_user_by_inbox?(user)
+            add_feedback_to_inbox(user)
+          end
+        end
+      end
+    end
+  end
+
+  def notify_user_of_own_comments?(user)
```

Comment:
It looks like this was a protected method when we were using an observer. Is it possible to have this (and the other formerly-protected methods) be protected here in the model? Or does moving them to the model mean we shouldn't worry about protecting them?

(Here's the full list of methods that were protected in the observer for quick reference: notify_parent_comment_owner(comment), have_different_owner?(comment, parent_comment), not_user_commenter?(parent_comment), content_too_different?(new_content, old_content), add_feedback_to_inbox(user, comment), update_feedback_in_inbox(user, comment), notify_user_by_email?(user), notify_user_by_inbox?(user), notify_user_of_own_comments?(user))